src/util.js

```diff
@@ -242,6 +242,23 @@ var util = {
     }
   },
 
+  /**
+   * Extract and parse JSON data from localStorage.
+   *
+   * @param {string} key - localStorage key
+   *
+   * @returns {object} parsed object or undefined
+   */
+  getJSONFromLocalStorage (key) {
+    const context = util.getGlobalContext();
+
+    try {
+      return JSON.parse(context.localStorage.getItem(key));
+    } catch(e) {
```

Comment:
Maybe an explicit `null` would be a better return value in this case.

However, both could be confusing, if someone edits their cached data manually and has a JSON error in it. Now that it's a seperate function, we could also actually check if the key exists and then make it throw an exception if it does, and contains data (not an empty string or sth), which is not JSON-formatted.

Reply:
This would change the functionality though. I'm not opposed to doing this, but this PR is just supposed to contain some refactorings that don't change any logic.

Should I create a separate issue for this?